tests/test_database_matrix.py

```diff
@@ -0,0 +1,571 @@
+from mock import call
+import pytest
+
+import nio
+
+from opsdroid.connector.matrix import ConnectorMatrix
+from opsdroid.core import OpsDroid
+from opsdroid.database.matrix import DatabaseMatrix
+
+from json import JSONEncoder
+
+
+@pytest.fixture
+def patched_send(mocker):
+    return mocker.patch("nio.AsyncClient._send")
+
+
+@pytest.fixture
+def opsdroid_matrix():
+    connector = ConnectorMatrix(
+        {
+            "rooms": {"main": "#test:localhost"},
+            "mxid": "@opsdroid:localhost",
+            "password": "hello",
+            "homeserver": "http://localhost:8008",
+        }
+    )
+    connector.room_ids = {"main": "!notaroomid"}
+    api = nio.AsyncClient("https://notaurl.com", None)
+    api.access_token = "arbitrarytoken"
+    connector.connection = api
+
+    with OpsDroid() as opsdroid:
+        opsdroid.connectors.append(connector)
+        yield opsdroid
+
+
+def matrix_call(method, path, content=None):
+    state_key = path.partition("dev.opsdroid.database/")[2]
+    path += "?access_token=arbitrarytoken"
+    if not content:
+        return call(
+            nio.RoomGetStateEventResponse,
+            method,
+            path,
+            response_data=("dev.opsdroid.database", f"{state_key}", "!notaroomid"),
+        )
+    else:
+        return call(
+            nio.RoomPutStateResponse,
+            method,
+            path,
+            JSONEncoder(separators=(",", ":")).encode(content),
+            response_data=("!notaroomid",),
+        )
+
+
+@pytest.mark.asyncio
+async def test_default_config(patched_send, opsdroid_matrix):
+    patched_send.return_value = nio.RoomGetStateEventResponse({}, "", "", "")
+
+    db = DatabaseMatrix({}, opsdroid=opsdroid_matrix)
+    db.should_migrate = False
+    await db.put("twim", {"hello": "world"})
+
+    patched_send.assert_has_calls(
+        [
+            matrix_call(
+                "GET",
+                "/_matrix/client/r0/rooms/%21notaroomid/state/dev.opsdroid.database/",
+            ),
```

Comment:
I think there is an issue with the tests here, that we should only be making this call once, but from the implementation I have to assume we are making it more than once. (Same goes for all the tests)